### PR TITLE
Add codefix that offers to add a name to an anonymous type member when one is missing.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/28928

Looks like this:

image

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @jcouv @jinujoseph

**Comment by CyrusNajmabadi** [comment]
@sharwell @jcouv can you please take a look?  Tahnks!

**Comment by CyrusNajmabadi** [comment]
Tagging @jcouv @jinujoseph

**Comment by CyrusNajmabadi** [comment]
@sharwell @jcouv can you please take a look? Tahnks!

**Comment by CyrusNajmabadi** [comment]
@jcouv @sharwell can you take a look?

**Comment by CyrusNajmabadi** [comment]
@sharwell @jcouv can you please take a look? Tahnks!

File: src/EditorFeatures/CSharpTest/AddAnonymousTypeMemberName/AddAnonymousTypeMemberNameTests.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+{
+    void M()
+    {
+        var v = new { Type = 1, {|Rename:Type1|} = this.GetType() };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddAnonymousTypeMemberName)]
+        public async Task TestFixAll1()
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/AddAnonymousTypeMemberName/AddAnonymousTypeMemberNameTests.cs]: 💡 Add a Fix All test for name collision:

```csharp
var v = new { this.GetType(), this.GetType() };
```

File: src/EditorFeatures/VisualBasicTest/AddAnonymousTypeMemberName/AddAnonymousTypeMemberNameTests.vb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+"
+class C
+    sub M()
+        dim v = new with {.v = 1, .{|Rename:V1|} = me.Equals(1)}
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddAnonymousTypeMemberName)>
+        Public Async Function TestFixAll1() As Task
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/AddAnonymousTypeMemberName/AddAnonymousTypeMemberNameTests.vb]: :bulb: Would like to see a name collision test here as well.

**Comment by CyrusNajmabadi** [comment]
@sharwell Tests added.  Feel free to merge when convenient.

**Comment by CyrusNajmabadi** [comment]
@sharwell @jinujoseph feel free to merge when convenient.

**Comment by jinujoseph** [comment]
test windows_debug_vs-integration_prtest

**Comment by CyrusNajmabadi** [comment]
@sharwell @jinujoseph feel free to merge in.

**Comment by MohammadHamdyGhanem** [comment]
@CyrusNajmabadi 
Well done. Thanks.

**Comment by MohammadHamdyGhanem** [comment]
@CyrusNajmabadi 
what about tuples:
```
var Res = from Control c in Controls
          orderby c.Name
          group c by (c.GetType(), c.Enabled);
```

This will not cause any error, but the Tuple definition will be
```(Type, bool Enabled)```

An analyzer can give the following suggestion:
> add an explicit name:

and suggests:
```C#
var Res = from Control c in Controls
          orderby c.Name
          group c by (Name1 = c.GetType(), c.Enabled);
```

so the tuple definition will be
```(Type Name1, bool Enabled)```

**Comment by CyrusNajmabadi** [comment]
> what about tuples:

You are welcome to contribute such a change.

**Pull request merged**